### PR TITLE
feat(gui): realease gui v1

### PR DESCRIPTION
**Pull request by wangxiaodong1**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: gui/backend/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# cubefs console
+
+**TODO**
+- [ ] dir adjust
+- [ ] api documents
+- [ ] blobstore api
```
Comment (wangxiaodong1) [inline gui/backend/README.md:6]: delete these todo

File: gui/frontend/vue.config.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+const { defineConfig } = require('@vue/cli-service')
+
+module.exports = defineConfig({
+  publicPath: '/portal',
+  transpileDependencies: true,
+  lintOnSave: false,
+  devServer: {
+    proxy: {
+      '/api': {
+        target: 'http://10.176.204.37:6007',
```
Comment (weilephp) [inline gui/frontend/vue.config.js:11]: 这里的ip去掉

File: gui/backend/service/auth/defaults.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+import (
+	"github.com/cubefs/cubefs/console/backend/model"
+)
+
+var (
+	DefaultRole = []model.AuthRole{
+		{RoleCode: "Root", RoleName: "root_user", MaxVolCount: -1, MaxVolSize: -1},
+		{RoleCode: "Ops", RoleName: "ops_user", MaxVolCount: -1, MaxVolSize: -1},
+		{RoleCode: "Normal", RoleName: "normal_user", MaxVolCount: 1, MaxVolSize: 2},
```
Comment (wangxiaodong1) [inline gui/backend/service/auth/defaults.go:11]: change the three RoleNames to:
admin_role
operator_role
viewer_role

**Comment by xiaochunhe** [comment]
I recommend putting the project into another repository cubefs-dashboard

File: gui/backend/handler/blobstore/blobstore.go [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+type OverviewOutput struct {
+	clustermgr.StatInfo
+	CodeModeInfo []common.CodeMode `json:"code_mode_info"`
+}
+
+func Overview(c *gin.Context) {
+	consulAddr, err := ginutils.GetConsulAddr(c)
+	if err != nil {
+		return
```
[the remark this answers is not in the document]
Comment (ahdda) [reply gui/backend/handler/blobstore/blobstore.go:25]: 有错误日志

File: gui/backend/handler/blobstore/conf/config.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+	Repair        []interface{} `json:"repair"`
+	Drop          []interface{} `json:"drop"`
+	Balance       []interface{} `json:"balance"`
+	ManualMigrate []interface{} `json:"manual_migrate"`
+	Inspect       []interface{} `json:"inspect"`
+	ShardRepair   []*TinkerData `json:"shard_repair"`
+	BlobDelete    []*TinkerData `json:"blob_delete"`
+}
+
+type TinkerData struct {
```
[the remark this answers is not in the document]
Comment (ahdda) [reply gui/backend/handler/blobstore/conf/config.go:25]: 改成task是否可以？

File: gui/backend/service/cluster/cluster.go [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+	output := httputils.Output{Data: &proto.ClusterView{}}
+	_, err = httputils.HandleResponse(c, req, err, &output)
+	if err != nil {
+		return nil, err
+	}
+	if output.Code != proto.ErrCodeSuccess {
+		return nil, errors.New(output.Msg)
+	}
+	return output.Data.(*proto.ClusterView), nil
+}
```
Comment (setcy) [inline gui/backend/service/cluster/cluster.go:29]: May I ask why you don't choose to use sdk here, but rewrite the http request? In other words, have you considered using the sdk?

File: gui/backend/README.md [hunk 7]
```diff
@@ -0,0 +1,3 @@
+# cubefs gui
+
+**TODO**
```
[the remark this answers is not in the document]
Comment (wangxiaodong1) [reply gui/backend/README.md:3]: done

File: gui/backend/handler/blobstore/blobstore.go [hunk 8]
```diff
@@ -0,0 +1,108 @@
+package blobstore
+
+import (
+	"github.com/cubefs/cubefs/blobstore/api/clustermgr"
```
[the remark this answers is not in the document]
Comment (wangxiaodong1) [reply gui/backend/handler/blobstore/blobstore.go:4]: done

File: gui/backend/handler/blobstore/common/disk.go [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+		if len(disks.Disks) == 0 {
+			break
+		}
+		out.Disks = append(out.Disks, disks.Disks...)
+		marker = uint32(disks.Marker)
+	}
+	return out, err
+}
+
+// FilterHistory *过滤初历史坏盘,过滤规则:host+path有多个的情况根据disk小到大排序,不要disk最大的
```
[the remark this answers is not in the document]
Comment (wangxiaodong1) [reply gui/backend/handler/blobstore/common/disk.go:160]: done

File: gui/backend/handler/blobstore/conf/config.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (wangxiaodong1) [reply gui/backend/handler/blobstore/conf/config.go:25]: done

File: gui/backend/handler/s3/file.go [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+	})
+	var err error
+	if req.Send() != nil {
+		err = req.Send()
+	}
+	if err != nil {
+		log.Errorf("s3.ListObjectsV2 failed.args:%+v,err:%+v", input, err)
+		ginutils.Send(c, codes.ThirdPartyError.Code(), err.Error(), nil)
+		return
+	}
```
[the remark this answers is not in the document]
Comment (wangxiaodong1) [reply gui/backend/handler/s3/file.go:58]: done

File: gui/backend/handler/blobstore/common/disk.go [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+		disks.Disks = FilterStatus(disks.Disks, uint8(input.Status))
+	}
+	if input.Readonly != 0 {
+		disks.Disks = FilterReadonly(disks.Disks, input.Readonly)
+	}
+	if input.DiskId != 0 {
+		disks.Disks = FilterDiskId(disks.Disks, input.DiskId)
+	}
+	if input.NodeData {
+		return disks, nil
```
[the remark this answers is not in the document]
Comment (wangxiaodong1) [reply gui/backend/handler/blobstore/common/disk.go:123]: 不需要

File: gui/backend/handler/blobstore/common/disk.go [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+		if _, ok := diskMap[v.Host+v.Path]; ok {
+			diskMap[v.Host+v.Path] = append(diskMap[v.Host+v.Path], v)
+		} else {
+			diskMap[v.Host+v.Path] = []*blobnode.DiskInfo{v}
+		}
+	}
+	for _, v := range diskMap {
+		if len(v) > 1 {
+			SortByDiskId(v)
+			output = append(output, v[:len(v)-1]...)
```
[the remark this answers is not in the document]
Comment (wangxiaodong1) [reply gui/backend/handler/blobstore/common/disk.go:174]: historical requirements

File: gui/backend/service/cluster/cluster.go (hunk 6 above)
Comment (wangxiaodong1) [reply gui/backend/service/cluster/cluster.go:29]: good idea. we consider use the sdk laster.

**Pull request closed without merging**